contracts/tokens/SecurityToken.sol

```diff
@@ -721,4 +738,62 @@ contract SecurityToken is StandardToken, DetailedERC20, ReentrancyGuard, Registr
         return _getValueAt(checkpointBalances[_investor], _checkpointId, balanceOf(_investor));
     }
 
+    /**
+     * @notice Use to get the list of controller addresses
+     * @return address array containing the list of controller addresses
+     */
+    function getControllers() public view returns (address[]) {
+        return controllers;
+    }
+
+    /**
+     * @notice Use by the issuer ot set the status of a controller addresses
+     * @param _controller address of the controller
+     * @param _status bool representing new status to set
+     */
+    function setControllerStatus(address _controller, bool _status) public onlyOwner {
+        require(isController[_controller] != _status);
+        if (_status == true) {
+            isController[_controller] = true;
+            controllerID[_controller] = controllers.length;
+            controllers.push(_controller);
+        } else {
+            isController[_controller] = false;
+            uint256 i = controllerID[_controller];
+            address last = controllers[controllers.length - 1];
+            controllers[i] = last;
+            controllerID[last] = i;
+
+            delete controllers[controllers.length - 1];
+            delete controllerID[_controller];
+            controllers.length--;
+        }
+        emit LogChangeControllerStatus(_controller, _status);
+    }
+
+    /**
+     * @notice Use by a controller to execute a foced transfer
+     * @param _from address from which to take tokens
+     * @param _to address where to send tokens
+     * @param _value amount of tokens to transfer
+     * @param _data data attached to the transfer by controller for event
+     */
+    function controllerTransfer(address _from, address _to, uint256 _value, bytes _data) public onlyController {
```

Comment:
Wouldn't this function also need to be added to the constructor?
transferFunctions[bytes4(keccak256(...))] = true;
@adamdossa

Reply:
Only if it calls `verifyTransfer()`

Reply:
This pattern can be revisited using internal / private functions

Reply:
Yes - if it is going to call `verifyTransfer` then it should do so that TMs know they should record any side-effects of the transfer.

One option here is to call `verifyTransfer` but disregard the result (or just include the result in the ControlledTransfer event).

Reply:
I like the idea of adding the result to the event